FerramAerospaceResearch.Base/UnityJobs/OcclusionJobs.cs

```diff
@@ -0,0 +1,381 @@
+                    projection = math.mul(math.inverse(quaternions[index]), projection);
+
+                    // The x,y coords of projection are again aligned to the vessel axes.
+                    min.x = math.min(min.x, projection.x);
+                    min.y = math.min(min.y, projection.y);
+                    max.x = math.max(max.x, projection.x);
+                    max.y = math.max(max.y, projection.y);
+                }
+            }
+
+            bounds[index] = new float2(math.abs(max.x - min.x), math.abs(max.y - min.y));
+        }
+    }
+
+    [BurstCompile]
+    public struct SetIntervalAndDimensionsJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<float2> bounds;
+        [ReadOnly] public int maxDim;
+        [ReadOnly] public float resolution;
+        [WriteOnly] public NativeArray<float2> interval;
+        [WriteOnly] public NativeArray<int2> dims;
+
+        public void Execute(int index)
+        {
+            int minXsize = (int)math.ceil(bounds[index].x / resolution);
+            int minYsize = (int)math.ceil(bounds[index].y / resolution);
+            dims[index] = new int2(math.min(maxDim, minXsize), math.min(maxDim, minYsize));
+
+            interval[index] = new float2(math.max(bounds[index].x / maxDim, resolution), math.max(bounds[index].y / maxDim, resolution));
+        }
+    }
+
+    [BurstCompile]
+    public struct SetVectorsJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<quaternion> quaternions;
+        [ReadOnly] public float4x4 localToWorldMatrix;
+        [WriteOnly] public NativeArray<float3> forward;
+        [WriteOnly] public NativeArray<float3> up;
+        [WriteOnly] public NativeArray<float3> right;
+
+        public void Execute(int index)
+        {
+            float3 dir = math.mul(quaternions[index], new float3(0, 0, 1));
+            float4 tmp = math.mul(localToWorldMatrix, new float4(dir.x, dir.y, dir.z, 0));
+            forward[index] = new float3(tmp.x, tmp.y, tmp.z);
+
+            dir = math.mul(quaternions[index], new float3(0, 1, 0));
+            tmp = math.mul(localToWorldMatrix, new float4(dir.x, dir.y, dir.z, 0));
+            up[index] = new float3(tmp.x, tmp.y, tmp.z);
+
+            dir = math.mul(quaternions[index], new float3(1 ,0, 0));
+            tmp = math.mul(localToWorldMatrix, new float4(dir.x, dir.y, dir.z, 0));
+            right[index] = new float3(tmp.x, tmp.y, tmp.z);
+        }
+    }
+
+    [BurstCompile]
+    public struct AngleBetween : IJobParallelFor
+    {
+        [ReadOnly] public float3 dir;
+        [ReadOnly] public NativeArray<quaternion> rotations;
+        [WriteOnly] public NativeArray<float> angles;
+
+        public void Execute(int index)
+        {
+            float3 rot_dir = math.mul(rotations[index], new float3(0, 0, 1));
+            angles[index] = math.degrees(math.acos(math.min(math.max(math.dot(dir, rot_dir), -1f), 1f)));
```

Comment:
`math.clamp` instead of `min` and `max` and is degrees really needed here when all trigonometric functions work with radians?

Reply:
Partially Implemented in 53809f6.  Left as degrees, pending reviewing how it gets used outside of the job.  (I'm certain I thought about this, and it's probably because some of the Physics calcs from RA use degrees.  So does..., much of Unity that isn't in Unity.Mathematics?)